File: server/http_test.js
const https = require('https');
const http = require('http');

console.log('Test des requêtes HTTP en Node.js:');

// Test avec le module http natif
function testNativeHttp() {
    console.log('\n1. Test avec http.get:');
    const req = http.get('http://example.com', (res) => {
        console.log('✓ Status:', res.statusCode);
        res.on('data', () => {});
        res.on('end', () => {
            testHttpsRequest();
        });
    });

    req.on('error', (error) => {
        console.log('✓ Erreur capturée:', error.message);
        testHttpsRequest();
    });
}

// Test avec https
function testHttpsRequest() {
    console.log('\n2. Test avec https.request:');
    const options = {
        hostname: 'api.github.com',
        path: '/users/octocat',
        method: 'GET',
        headers: {
            'User-Agent': 'Node.js Test'
        }
    };

    const req = https.request(options, (res) => {
        console.log('✓ Status:', res.statusCode);
        res.on('data', () => {});
        res.on('end', () => {
            testFetch();
        });
    });

    req.on('error', (error) => {
        console.log('✓ Erreur capturée:', error.message);
        testFetch();
    });

    req.end();
}

// Test avec fetch API
async function testFetch() {
    console.log('\n3. Test avec fetch:');
    try {
        const response = await fetch('https://jsonplaceholder.typicode.com/posts/1');
        console.log('✓ Status:', response.status);
    } catch (error) {
        console.log('✓ Erreur capturée:', error.message);
    }
}

// Démarrer les tests
testNativeHttp(); 

File: server/http_test.py
import urllib.request
import urllib.error
import http.client
import requests  # Cette importation devrait être bloquée

print('Test des requêtes HTTP en Python:')

def test_urllib():
    print('\n1. Test avec urllib:')
    try:
        with urllib.request.urlopen('http://example.com') as response:
            print('✓ Status:', response.status)
    except urllib.error.URLError as e:
        print('✓ Erreur capturée:', str(e))
    except Exception as e:
        print('✓ Autre erreur:', str(e))

def test_http_client():
    print('\n2. Test avec http.client:')
    try:
        conn = http.client.HTTPSConnection('api.github.com')
        conn.request('GET', '/users/octocat', headers={'User-Agent': 'Python Test'})
        response = conn.getresponse()
        print('✓ Status:', response.status)
        conn.close()
    except Exception as e:
        print('✓ Erreur capturée:', str(e))

def test_requests():
    print('\n3. Test avec requests:')
    try:
        response = requests.get('https://jsonplaceholder.typicode.com/posts/1')
        print('✓ Status:', response.status_code)
    except Exception as e:
        print('✓ Erreur capturée:', str(e))

# Exécuter les tests
test_urllib()
test_http_client()
test_requests() 